checks.py

```diff
@@ -0,0 +1,414 @@
+
+            #: Static items should be tagged 'Static'
+            if self.static_shelved == 'static':
+                if 'Static' not in self.new_tags:
+                    self.new_tags.append('Static')
+                if 'Shelved' in self.new_tags:
+                    self.new_tags.remove('Shelved')
+
+            #: Make sure it's got SGID, AGRC in it's tags
+            if 'SGID' not in self.new_tags:
+                self.new_tags.append('SGID')
+            if 'AGRC' not in self.new_tags:
+                self.new_tags.append('AGRC')
+
+        #: Create tags data: tags_fix, tags_old, tags_new
+        if sorted(self.new_tags) != sorted(self.item.tags):
+            tags_data = {'tags_fix':'Y', 'tags_old':self.item.tags, 'tags_new':self.new_tags}
+        else:
+            tags_data = {'tags_fix':'N', 'tags_old':'', 'tags_new':''}
+
+        self.results_dict.update(tags_data)
+
+
+    def title_check(self):
+        '''
+        Check item's title against title in metatable.
+
+        Update results_dict with results for this item:
+                {'title_fix':'', 'title_old':'', 'title_new':''}
+        '''
+
+        #: Create title data: title_fix, title_old, title_new
+        #: Always include the old title for readability
+        if self.new_title and self.new_title != self.item.title:
+            title_data = {'title_fix':'Y', 'title_old':self.item.title, 'title_new':self.new_title}
+        else:
+            title_data = {'title_fix':'N', 'title_old':self.item.title, 'title_new':''}
+
+        self.results_dict.update(title_data)
+
+
+    def folder_check(self, itemid_and_folder):
+        '''
+        Check item's folder against SGID category name from metatable.
+
+        Update results_dict with results for this item:
+                {'folder_fix':'', 'folder_old':'', 'folder_new':''}
+        '''
+
+        #: Get current folder from dictionary of items' folders
+        current_folder = itemid_and_folder[self.item.itemid]
+
+        #: Create folder data: folder_fix, folder_old, folder_new
+        if self.new_folder and self.new_folder != current_folder:
+            folder_data = {'folder_fix':'Y', 'folder_old':current_folder, 'folder_new':self.new_folder}
+        else:
+            folder_data = {'folder_fix':'N', 'folder_old':'', 'folder_new':''} 
+
+        self.results_dict.update(folder_data)
+
+
+    def groups_check(self):
+        '''
+        Check item's group against SGID category from metatable.
+
+        Update results_dict with results for this item:
+                {'groups_fix':'', 'groups_old':'', 'group_new':''}
+        '''
+
+        #: Get current group, wrapped in try/except for groups that error out
+        try:
+            current_groups = [group.title for group in self.item.shared_with['groups']]
+        except:
+            current_groups = ['Error']
+
+        #: Create groups data: groups_fix, groups_old, group_new
+        if current_groups == 'Error':
```

Comment:
it might be safer to do string checks without case sensitivity. i'll trust that this works now but can you guarantee the casing? Is that something that you want to have to think about or worry about in the future? I would always recommend case insensitive string comparisons.

Reply:
TIL str.casefold() is more appropriate than str.lower(). Overkill for this case (where the program sets the 'Error' string) but I still added it to get familiar with casefold().